File: markdowner-filter/src/main/java/org/jesperancinha/parser/markdowner/filter/GradleFilter.java
package org.jesperancinha.parser.markdowner.filter;

import lombok.extern.slf4j.Slf4j;

import java.nio.file.Path;
import java.util.List;

/**
 * Filter to check if folder contains a Gradle project and keeps the project name in memory
 */
@Slf4j
public class GradleFilter extends ProjectFilter<Path> {

    private static final List<String> gradleBuilds = List.of(
            "gradle.build", "build.gradle");

    @Override
    public boolean test(Path path) {
        boolean isGradleBuild = gradleBuilds.contains(path.getFileName().toString());
        if (isGradleBuild) {
            this.lastProjectName = path.getParent().getFileName().toString();
            return true;
        }
        return false;
    }

    @Override
    public String lastProjectName() {
        return lastProjectName;
    }
}

File: markdowner-filter/src/main/kotlin/org/jesperancinha/parser/markdowner/filter/GradleFilter.kt
package org.jesperancinha.parser.markdowner.filter

import java.nio.file.Path

/**
 * Filter to check if folder contains a Gradle project and keeps the project name in memory
 */
class GradleFilter : ProjectFilter<Path>() {
    override fun test(path: Path): Boolean {
        val isGradleBuild = gradleBuilds.contains(path.fileName.toString())
        if (isGradleBuild) {
            lastProjectName = path.parent.fileName.toString()
            return true
        }
        return false
    }

    override fun lastProjectName(): String? {
        return lastProjectName
    }

    companion object {
        private val gradleBuilds = listOf(
            "gradle.build", "build.gradle", "build.gradle.kts"
        )
    }
}